mos_tests/ironic/manage_instances_test.py

```diff
@@ -88,3 +89,26 @@ def is_instance_active():
 
         assert getattr(os_conn.nova.servers.get(instance.id),
                        "OS-EXT-STS:vm_state") == 'active'
+
+
+@pytest.mark.check_env_('has_ironic_conductor')
+@pytest.mark.need_devops
+@pytest.mark.testrail_id('631916')
+def test_boot_instance_with_user_data(ubuntu_image, flavors, keypair,
+                                      ironic, ironic_nodes, os_conn, env):
+    """Boot Ubuntu14-based virtual-bare-metal instance with user-data
+
+    Scenario:
+        1. Boot ironic instance with user data file user_data.sh
+        2. Check that user_data.sh is present on instance
+        3. Ping 8.8.8.8 from instance
+    """
+    instance = ironic.boot_instance(image=ubuntu_image,
+                                    flavor=flavors[0],
+                                    keypair=keypair,
+                                    userdata='fake_command_to_test')
```

Comment:
it`s not clear for me, where we create userdata file before putting it into instance